File: subarray-sums-divisible-by-k/subarray-sums-divisible-by-k.cpp
class Solution {
public:
    int subarraysDivByK(vector<int>& nums, int k) {
        int n=nums.size(),ans=0,prefix_sum=0;
        vector<int>reminder(k);
        reminder[0]=1;
        for(int i=0;i<n;i++){
            prefix_sum=(prefix_sum+10000*k+nums[i])%k;
            ans+=reminder[prefix_sum]++;
        }
        return ans;
    }
   
    
};

File: 1016-subarray-sums-divisible-by-k/subarray-sums-divisible-by-k.py
class Solution:
    def subarraysDivByK(self, nums: List[int], k: int) -> int:
        mp = defaultdict(int)
        mp[0] = 1
        ans = 0
        sum = 0
        for i in range(len(nums)):
            sum+=nums[i]
            mod_val = sum%k 
            ans += mp[mod_val]
            mp[mod_val]+=1
        return ans